In [60]:
import pandas as pd

In [61]:
df1 = pd.read_csv("cit_015_urban_heat_island_effect.csv")
df1.head()

the_geom  es00pop isourbid iso3  \
0  0106000020E61000000700000001030000000100000009...    98097    FJI76  FJI   
1  0106000020E61000000100000001030000000100000016...    26800   RUS524  RUS   
2  0106000020E61000000500000001030000000200000066...   426550   RUS542  RUS   
3  0106000020E61000001000000001030000000100000006...    46164   FIN862  FIN   
4  0106000020E61000000900000001030000000100000009...   174294  ISL1285  ISL   

   urbid       name      schnm  es90pop   es95pop  cartodb_id  sqkm_final  \
0     76       Suva       SUVA    87419   92593.0          13   234.22200   
1    524    Dudinka    DUDINKA    27147   27105.0         138     8.56631   
2    542   Murmansk   MURMANSK   504818  463715.0         142  1270.26000   
3    862       Kemi       KEMI    46389   46238.0         178   930.25400   
4   1285  Reykjavík  REYKJAVIK   144117  158067.0         234   454.47200   

   urb_d_mean  buf_d_mean  d_t_diff  urb_n_mean  buf_n_mean  n_t_diff  \
0   33.063706   30.659449  2.404257   20.787647   20.275314  0.512333   
1   27.415333   28.350921 -0.935588   10.603333    8.706316  1.897018   
2   21.951042   22.412175 -0.461133    5.398211    5.115498  0.282714   
3   20.930596   21.392590 -0.461993    6.568162    5.620239  0.947924   
4   24.295258   24.872159 -0.576901    0.712965    0.605435  0.107529   

    latitude   longitude  
0 -18.069090  178.499678  
1  69.405123   86.178333  
2  68.966790   33.155192  
3  65.923008   24.376490  
4  64.117851  -21.774800

In [64]:
df1.sort_values('d_t_diff', ascending = False, inplace = True)
highest = df1.head(10)
highest
#highest.to_csv('highest_diff_temp')

the_geom  es00pop  isourbid  \
1171   0106000020E6100000010000000103000000010000001D...    26996  IND52489   
27607  0106000020E61000000100000001030000000100000025...    11610  IND90831   
16806  0106000020E6100000010000000103000000010000001B...    25225  IND52511   
3145   0106000020E6100000010000000103000000010000000D...    20370  IND54144   
27645  0106000020E61000000100000001030000000100000025...    20909  IND90850   
18565  0106000020E61000000100000001030000000100000025...    18073  ECU58519   
16390  0106000020E6100000010000000103000000010000002D...     2726  USA51110   
26019  0106000020E61000000100000001030000000100000025...     7263  IND90281   
15332  0106000020E61000000100000001030000000100000031...    55566  IND47966   
8867   0106000020E61000000100000001030000000100000017...     4517  KGZ28878   

      iso3  urbid             name            schnm  es90pop  es95pop  \
1171   IND  52489          JAMKHED          JAMKHED    20739  23777.0   
27607  IND  90831  HEGGADDEVANKOTE  HEGGADDEVANKOTE     8057   9672.0   
16806  IND  52511           MUKHED           MUKHED    19107  21954.0   
3145   IND  54144          HALIYAL          HALIYAL    17759  19020.0   
27645  IND  90850   NAMAGIRIPETTAI   NAMAGIRIPETTAI    16062  18416.0   
18565  ECU  58519      Pedro Carbo       PEDROCARBO    13783  16037.0   
16390  USA  51110       Kaunakakai       KAUNAKAKAI     2658   2692.0   
26019  IND  90281      TITABORTOWN      TITABORTOWN     5579   6397.0   
15332  IND  47966            DABRA            DABRA    45684  50383.0   
8867   KGZ  28878         Kajy-Say          KAJYSAY     3861   4123.0   

       cartodb_id  sqkm_final  urb_d_mean  buf_d_mean   d_t_diff  urb_n_mean  \
1171        15770    26.83390   57.056061   40.351035  16.705026   25.876061   
27607       27565    21.41230   46.841538   31.331615  15.509924   22.475385   
16806       15775    21.14320   55.118462   40.967592  14.150870   17.803077   
3145        16288     5.79668   43.612857   30.221803  13.391055   21.981429   
27645       27603    31.61330   50.522222   37.622408  12.899814   20.874211   
18565       17965    46.33400   43.030000   30.162296  12.867704   19.784510   
16390       15274    56.87590   46.801304   34.083725  12.717579   20.268485   
26019       25954    16.67410   35.080833   22.823682  12.257151   19.180000   
15332       14045    74.92870   49.233711   37.322199  11.911512   21.276186   
8867         6277    16.10760   42.197692   31.207069  10.990623   12.933846   

       buf_n_mean  n_t_diff   latitude   longitude  
1171    23.959950  1.916111  18.739091   75.318047  
27607   21.664349  0.811036  12.062458   76.297172  
16806   18.678412 -0.875335  18.710264   77.376270  
3145    20.636867  1.344562  15.335448   74.770824  
27645   21.735763 -0.861553  11.466669   78.266670  
18565   20.036609 -0.252099  -1.813405  -80.225931  
16390   19.095269  1.173216  21.092289 -157.007337  
26019   19.470860 -0.290860  26.600001   94.199997  
15332   21.364503 -0.088317  25.888688   78.338693  
8867    12.493034  0.440812  42.146163   77.188687

# Results
Out of the 10 cities that experience the highest urban heat island effect, 7 of them are in India (Jamkhed, Heggaddevankote, Mukhed, Haliyal, Namagiripettai, Titabortown, and Dabra). The others in the top 10 include Pedro Carbo, Ecuador, Kaunakakai, HI, USA, and Kajy-Say, Kyrgyzstan.  

In [46]:
df1.columns

Index(['the_geom', 'es00pop', 'isourbid', 'iso3', 'urbid', 'name', 'schnm',
       'es90pop', 'es95pop', 'cartodb_id', 'sqkm_final', 'urb_d_mean',
       'buf_d_mean', 'd_t_diff', 'urb_n_mean', 'buf_n_mean', 'n_t_diff',
       'latitude', 'longitude'],
      dtype='object')

In [49]:
df1.d_t_diff.describe()

count    31500.000000
mean         0.731544
std          1.851282
min        -30.909236
25%         -0.131641
50%          0.696797
75%          1.638193
max         16.705026
Name: d_t_diff, dtype: float64

In [50]:
#group by country
countries = df1.groupby('iso3').describe()
countries.d_t_diff
diff = countries.d_t_diff
diff.sort_values('mean', ascending = False)

count      mean       std       min       25%       50%       75%  \
iso3                                                                      
MLT     2.0  7.073437  1.100096  6.295552  6.684494  7.073437  7.462379   
GGY     2.0  5.686007  2.019051  4.258322  4.972164  5.686007  6.399849   
ANT     1.0  5.635766       NaN  5.635766  5.635766  5.635766  5.635766   
ABW     1.0  5.319243       NaN  5.319243  5.319243  5.319243  5.319243   
JEY     1.0  4.746170       NaN  4.746170  4.746170  4.746170  4.746170   
BHR     1.0  4.348915       NaN  4.348915  4.348915  4.348915  4.348915   
TCA     2.0  4.338519  1.094564  3.564545  3.951532  4.338519  4.725506   
TWN     8.0  4.099386  2.229468  0.626158  2.672898  4.043805  5.832749   
BRB     1.0  3.934619       NaN  3.934619  3.934619  3.934619  3.934619   
BMU     1.0  2.873558       NaN  2.873558  2.873558  2.873558  2.873558   
MKD    21.0  2.804958  1.816435 -1.028668  1.667835  2.335694  3.556266   
CYM     1.0  2.638455       NaN  2.638455  2.638455  2.638455  2.638455   
FRO     3.0  2.604174  1.286213  1.747047  1.864692  1.982336  3.032737   
BIH    24.0  2.563965  1.059421  0.666851  1.886488  2.386574  3.210249   
AIA     1.0  2.449857       NaN  2.449857  2.449857  2.449857  2.449857   
JPN   144.0  2.437923  1.772457 -5.718522  1.209923  2.411886  3.435190   
SJM     1.0  2.387752       NaN  2.387752  2.387752  2.387752  2.387752   
ECU    41.0  2.353433  3.640487 -3.088699  0.002988  1.047479  4.509002   
GRL    14.0  2.318307  2.323633 -1.307514  0.393249  2.639022  4.438227   
GEO    54.0  2.297282  1.816934 -1.271390  1.008838  2.312640  3.513021   
KOR    52.0  2.214665  0.929500  0.535731  1.730493  2.101772  2.757088   
SVN    65.0  2.156971  1.404598 -0.290081  1.085278  2.191539  3.009862   
CHE    72.0  2.127962  1.779753 -1.012204  0.988012  1.820141  2.898896   
WSM     8.0  2.087125  1.515688 -0.771314  1.536836  2.241275  2.992647   
SGP     1.0  1.912663       NaN  1.912663  1.912663  1.912663  1.912663   
VUT     2.0  1.903236  0.665979  1.432318  1.667777  1.903236  2.138695   
MUS     6.0  1.858419  2.070590 -1.300686  0.598546  2.396088  3.219919   
GTM    29.0  1.846987  1.603795 -1.359219  0.666075  2.262358  2.639074   
SVK   102.0  1.846836  1.269629 -3.141450  1.121398  1.891322  2.525405   
NCL     8.0  1.842500  1.715901 -0.772147  0.391696  2.154465  3.419018   
...     ...       ...       ...       ...       ...       ...       ...   
TUN    57.0 -0.053470  1.813954 -4.291854 -1.080697 -0.269339  0.780788   
SWZ    13.0 -0.127841  2.060352 -2.290643 -1.794071 -0.566819  0.436564   
KGZ    80.0 -0.140208  3.409946 -6.753162 -2.600623 -0.219742  2.455525   
QAT     6.0 -0.147464  1.048451 -1.598402 -0.606442 -0.267057  0.292617   
ISR     8.0 -0.149319  1.159556 -1.959818 -0.844358 -0.291253  0.509218   
SYC     1.0 -0.158872       NaN -0.158872 -0.158872 -0.158872 -0.158872   
LBN    11.0 -0.166273  1.423075 -2.197131 -1.452157  0.401738  0.766457   
AND     1.0 -0.177644       NaN -0.177644 -0.177644 -0.177644 -0.177644   
SPM     2.0 -0.209564  0.219627 -0.364864 -0.287214 -0.209564 -0.131914   
CYP     4.0 -0.225694  1.886355 -2.655880 -1.203270  0.080962  1.058537   
IRN   136.0 -0.259292  1.961661 -5.845221 -1.417005 -0.082049  0.901368   
SYR    27.0 -0.273006  1.063335 -3.032007 -0.791214 -0.287040  0.188992   
TJK    17.0 -0.280296  3.195467 -5.187888 -2.471361 -0.123289  2.025908   
MMR    60.0 -0.304094  2.343672 -7.012612 -1.616434 -0.394102  0.682878   
LBY    31.0 -0.320234  1.074302 -2.122145 -1.420389 -0.012294  0.469280   
ISL    20.0 -0.357369  1.623836 -4.672695 -1.154953 -0.305860  0.512317   
SMR     1.0 -0.365267       NaN -0.365267 -0.365267 -0.365267 -0.365267   
MAR    70.0 -0.426210  1.479535 -4.616377 -1.118863 -0.403774  0.676886   
YEM    31.0 -0.442746  1.888394 -5.994758 -1.080738 -0.153675  0.424503   
MAC     1.0 -0.749915       NaN -0.749915 -0.749915 -0.749915 -0.749915   
ARE     3.0 -0.763946  0.323759

In [51]:
df1.set_index("name", inplace=True)

In [52]:
df1.loc['Karachi']

the_geom      0106000020E61000000800000001030000000100000006...
es00pop                                                 9919729
isourbid                                               PAK48573
iso3                                                        PAK
urbid                                                     48573
schnm                                                   KARACHI
es90pop                                                 7066873
es95pop                                             8.37266e+06
cartodb_id                                                14274
sqkm_final                                              1846.04
urb_d_mean                                              39.5701
buf_d_mean                                              36.6438
d_t_diff                                                 2.9263
urb_n_mean                                              21.5418
buf_n_mean                                              19.9638
n_t_diff                                

In [53]:
#Import WRI countries table
countries = pd.read_excel('countries.xls')
countries.head()

ADM0_A3  abbrev continent                    formal_nam iso_a2 iso_a3  \
0     AFG    Afg.      Asia  Islamic State of Afghanistan     AF    AFG   
1     AGO    Ang.    Africa   People's Republic of Angola     AO    AGO   
2     ALB    Alb.    Europe           Republic of Albania     AL    ALB   
3     AND    And.    Europe       Principality of Andorra     AD    AND   
4     ARE  U.A.E.      Asia          United Arab Emirates     AE    ARE   

   iso_n3                   iso_short                  name  \
0       4                 Afghanistan           Afghanistan   
1      24                      Angola                Angola   
2       8                     Albania               Albania   
3      20                     Andorra               Andorra   
4     784  United Arab Emirates (the)  United Arab Emirates   

              name_sort             ...              \
0           Afghanistan             ...               
1                Angola             ...               
2               Albania             ...               
3               Andorra             ...               
4  United Arab Emirates             ...               

                          un_es                      un_fr un_n3 un_region  \
0               Afganistán (el)   Afghanistan (l') [masc.]     4      Asia   
1                        Angola        Angola (l') [masc.]    24    Africa   
2                       Albania        Albanie (l') [fém.]     8    Europe   
3                       Andorra        Andorre (l') [fém.]    20    Europe   
4  Emiratos Árabes Unidos (los)  Émirats arabes unis (les)   784      Asia   

                           un_ru       un_subregi     un_zh wb_a2 wb_a3  \
0                     Афганистан    Southern Asia       阿富汗    AF   AFG   
1                         Ангола    Middle Africa       安哥拉    AO   AGO   
2                        Албания  Southern Europe     阿尔巴尼亚    AL   ALB   
3                        Андорра  Southern Europe       安道尔    AD   ADO   
4  Объединенные Арабские Эмираты     Western Asia  阿拉伯联合酋长国    AE   ARE   

                    wb_region  
0                  South Asia  
1          Sub-Saharan Africa  
2       Europe & Central Asia  
3       Europe & Central Asia  
4  Middle East & North Africa  

[5 rows x 28 columns]

In [54]:
#Join countries and heat island by countries to sort by regions and make understand ISO names
country_heat = pd.merge(diff, countries, how='inner', right_on= 'iso_a3', left_index= True)

In [55]:
country_heat.sort_values('mean', ascending = False)

count      mean       std       min       25%       50%       75%  \
115    2.0  7.073437  1.100096  6.295552  6.684494  7.073437  7.462379   
17     1.0  4.348915       NaN  4.348915  4.348915  4.348915  4.348915   
24     1.0  3.934619       NaN  3.934619  3.934619  3.934619  3.934619   
113   21.0  2.804958  1.816435 -1.028668  1.667835  2.335694  3.556266   
19    24.0  2.563965  1.059421  0.666851  1.886488  2.386574  3.210249   
86   144.0  2.437923  1.772457 -5.718522  1.209923  2.411886  3.435190   
50    41.0  2.353433  3.640487 -3.088699  0.002988  1.047479  4.509002   
62    54.0  2.297282  1.816934 -1.271390  1.008838  2.312640  3.513021   
93    52.0  2.214665  0.929500  0.535731  1.730493  2.101772  2.757088   
162   65.0  2.156971  1.404598 -0.290081  1.085278  2.191539  3.009862   
30    72.0  2.127962  1.779753 -1.012204  0.988012  1.820141  2.898896   
188    8.0  2.087125  1.515688 -0.771314  1.536836  2.241275  2.992647   
152    1.0  1.912663       NaN  1.912663  1.912663  1.912663  1.912663   
187    2.0  1.903236  0.665979  1.432318  1.667777  1.903236  2.138695   
121    6.0  1.858419  2.070590 -1.300686  0.598546  2.396088  3.219919   
70    29.0  1.846987  1.603795 -1.359219  0.666075  2.262358  2.639074   
161  102.0  1.846836  1.269629 -3.141450  1.121398  1.891322  2.525405   
26    18.0  1.817430  4.758046 -9.577325  0.511579  2.821024  4.432699   
9    126.0  1.807251  1.381611 -2.181326  0.891764  1.721871  2.650338   
16    37.0  1.792504  2.255356 -4.241239  0.198527  1.917915  3.205914   
100    1.0  1.792280       NaN  1.792280  1.792280  1.792280  1.792280   
141   24.0  1.760379  1.518470 -3.075795  0.974897  1.626199  2.845483   
72    47.0  1.734817  1.793701 -2.209072  0.353808  1.629114  3.125270   
6     40.0  1.733350  2.683149 -3.377821  0.127735  1.512646  3.284565   
48    31.0  1.696676  1.305399 -0.207754  0.746118  1.235497  2.577871   
38    36.0  1.659527  2.407297 -3.653266  0.237058  2.023514  3.032406   
174    4.0  1.572130  1.903584 -1.114028  1.025569  2.047184  2.593745   
68   389.0  1.555974  2.047869 -5.135381  0.299100  1.491353  2.807326   
132   97.0  1.555244  1.604995 -4.169595  0.491854  1.549214  2.662832   
69     4.0  1.546966  1.239645 -0.261729  1.264195  1.989374  2.272145   
..     ...       ...       ...       ...       ...       ...       ...   
59     9.0  0.101985  0.791894 -1.212009 -0.192809  0.454716  0.554222   
49   374.0  0.095828  1.552212 -6.021947 -0.672687  0.200496  1.004054   
124   32.0  0.063284  1.405034 -4.074082 -0.813181 -0.046939  0.749377   
87    31.0 -0.043333  2.471746 -3.705191 -1.764541 -0.418638  0.465562   
175   57.0 -0.053470  1.813954 -4.291854 -1.080697 -0.269339  0.780788   
164   13.0 -0.127841  2.060352 -2.290643 -1.794071 -0.566819  0.436564   
89    80.0 -0.140208  3.409946 -6.753162 -2.600623 -0.219742  2.455525   
144    6.0 -0.147464  1.048451 -1.598402 -0.606442 -0.267057  0.292617   
82     8.0 -0.149319  1.159556 -1.959818 -0.844358 -0.291253  0.509218   
165    1.0 -0.158872       NaN -0.158872 -0.158872 -0.158872 -0.158872   
96    11.0 -0.166273  1.423075 -2.197131 -1.452157  0.401738  0.766457   
3      1.0 -0.177644       NaN -0.177644 -0.177644 -0.177644 -0.177644   
42     4.0 -0.225694  1.886355 -2.655880 -1.203270  0.080962  1.058537   
79   136.0 -0.259292  1.961661 -5.845221 -1.417005 -0.082049  0.901368   
166   27.0 -0.273006  1.063335 -3.032007 -0.791214 -0.287040  0.188992   
170   17.0 -0.280296  3.195467 -5.187888 -2.471361 -0.123289  2.025908   
116   60.0 -0.304094  2.343672 -7.012612 -1.616434 -0.394102  0.682878   
98    31.0 -0.320234  1.074302 -2.122145 -1.420389 -0.012294  0.469280   
81    20.0 -0.357369  1.623836 -4.672695 -1.154953 -0.305860  0.512317   
156    1.0 -0.365267       NaN -0.365267 -0.365267 -0.365267 -0.365267   
106   70.0 -0.426210  1.479535 -4.616377 -1.118863 -0.403774  0.676886   
189   31.0 -0.442746  1.888394 -5.994758 -1.080738 -0.153675  0.424503   
4      3.

# Results

Malta experiences the highest heat island effect at 7.07 degrees C above the buffer followed by Bahrein and Barbados. However this mean may not indicate which countries have significant issues, because other cities experience this effect with temperature differences that are double these average, but because of the includion of many more cities in that country, they do not show up significantly when aggregating at the country level. 

In [59]:
country_heat.groupby('wb_region').mean().sort_values('mean', ascending = False)

count      mean       std        min  \
wb_region                                                                
Latin America & Caribbean    113.333333  1.115475  1.664621  -2.303369   
East Asia & Pacific          267.241379  0.999922  1.413439  -2.432628   
Europe & Central Asia        141.320000  0.870624  1.515111  -2.477135   
North America               2811.500000  0.811789  1.338903  -6.381752   
Sub-Saharan Africa            62.404255  0.588520  1.611232  -3.102864   
Middle East & North Africa    44.900000  0.367957  1.395675  -2.509436   
South Asia                   469.428571  0.362421  3.023752 -14.763925   

                                 25%       50%       75%        max  \
wb_region                                                             
Latin America & Caribbean   0.200870  1.126581  1.997644   4.508956   
East Asia & Pacific         0.281989  0.966799  1.756465   3.937055   
Europe & Central Asia       0.010665  0.848771  1.659778   4.825210   
North America               0.128302  0.684714  1.401806  10.634567   
Sub-Saharan Africa         -0.341910  0.519640  1.429076   4.564254   
Middle East & North Africa -0.243058  0.505463  1.031414   2.903463   
South Asia                 -0.653412  0.656445  1.966685   7.432001   

                                iso_n3  primary       un_n3  
wb_region                                                    
Latin America & Caribbean   364.727273      1.0  364.727273  
East Asia & Pacific         452.137931      1.0  452.137931  
Europe & Central Asia       429.020000      1.0  429.020000  
North America               482.000000      1.0  482.000000  
Sub-Saharan Africa          451.446809      1.0  451.446809  
Middle East & North Africa  496.950000      1.0  496.950000  
South Asia                  246.857143      1.0  246.857143

# Results
On average Latin American and the Carribbean experience the highest urban heat island effect at 1.115 deg. C above the surrounding buffer, followed by East Asia & the Pacific at ~1.0 deg C.  